src/seedu/addressbook/commands/Command.java

```diff
@@ -38,8 +39,11 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
 
     /**
      * Executes the command and returns the result.
+     * @throws DuplicatePersonException 
      */
-    public abstract CommandResult execute();
+    public abstract CommandResult execute() throws DuplicatePersonException;
+    
+    public abstract boolean isMutating();
```

Comment:
It is better to mention it returns true if the Command *potentially* mutates the data (header comments specify a contract, so try to be as precise as possible). E.g. Adding a duplicate person will not mutate data but the corresponding AddCommand object will still return true for this method.